### ISSUE TITLE
Deadlocking issue when registering bgfetch

### ISSUE THREAD

**Issue by jakearchibald** [WICG/background-fetch#92]
If I add a request that has a stream for a body, but that stream never ends, it blocks additional bg fetches being registered until the stream provider is terminated.

As a workaround, the ID of the bg fetch could be reserved, but will allow fetches with different IDs to be registered in parallel.

No browser supports requests with stream bodies right now, so this issue is low priority.
